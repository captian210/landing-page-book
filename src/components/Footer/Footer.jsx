import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Social from '../Social';
import Copyright from '../Copyright';

class Footer extends Component {
  constructor(props) {
    super(props);
  }

  renderSocials() {
    const { socials, socialRounded, socialUrl, socialGray } = this.props;
    return socials.map((social, index) => (
      <Social key={index} type={social} gray={socialGray} rounded={socialRounded} url={socialUrl} />
    ));
  }

  renderSections() {
    const { sections } = this.props;
    return sections.map((section, index) => (
      <div className="col-sm-12 col-md text-center" key={index}>
        <div className="Footer__Sections__Header">{section.name}</div>
        {section.sections.length > 0 && (
          <ul className="list-unstyled text-small">{this.renderLinks(section.sections)}</ul>
        )}
      </div>
    ));
  }

  renderLinks(links) {
    return links.map((link, index) => (
      <li key={index}>
        <a className="Footer__Sections__Link" href={link.url}>
          {link.name}
        </a>
      </li>
    ));
  }

  render() {
    const { sections, socials, copyright, children } = this.props;
    return (
      <footer className="Footer p-5">
        {sections.length > 0 && <div className="row Footer__Sections">{this.renderSections()}</div>}
        {socials.length > 0 && <div className="mt-3">{this.renderSocials()}</div>}
        {copyright && (
          <div className="mt-3 Footer__Copyright">
            <Copyright text={copyright} />
          </div>
        )}
        {children && <React.Fragment>{children}</React.Fragment>}
      </footer>
    );
  }
}

Footer.propTypes = {
  socialRounded: PropTypes.bool,
  socialGray: PropTypes.bool,
  socialUrl: PropTypes.string,
  copyright: PropTypes.string,
  socials: PropTypes.array,
  sections: PropTypes.array
};
Footer.defaultProps = {
  socialRounded: true,
  socialGray: false,
  socialUrl: '',
  copyright: '',
  socials: [],
  sections: []
};

export default Footer;
