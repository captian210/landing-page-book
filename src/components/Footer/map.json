{"description":"","displayName":"Footer","methods":[{"name":"renderSocials","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"renderSections","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"renderLinks","docblock":null,"modifiers":["static"],"params":[{"name":"links","type":null}],"returns":null}],"props":{"socialRounded":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"socialGray":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"false","computed":false}},"socialUrl":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"''","computed":false}},"copyright":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"''","computed":false}},"socials":{"type":{"name":"arrayOf","value":{"name":"string"}},"required":false,"description":"","defaultValue":{"value":"[]","computed":false}},"sections":{"type":{"name":"arrayOf","value":{"name":"shape","value":{"name":{"name":"string","required":false},"sections":{"name":"arrayOf","value":{"name":"shape","value":{"name":{"name":"string","required":false},"url":{"name":"string","required":false}}},"required":false}}}},"required":false,"description":"","defaultValue":{"value":"[]","computed":false}},"children":{"type":{"name":"union","value":[{"name":"arrayOf","value":{"name":"node"}},{"name":"node"}]},"required":false,"description":"","defaultValue":{"value":"null","computed":false}}},"filename":"/components/Footer/Footer.jsx"}