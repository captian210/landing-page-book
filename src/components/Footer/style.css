:root {
  --Footer-color: #cccccc;
  --Footer-backgroundColor: #383d45;
  --Footer__Sections__Link--hover-color: #fed136;
  --Footer__Sections__Header-fontSize: 1.5rem;
  --Footer__Sections__Link-fontSize: 0.8rem;
  --Footer__Copyright-fontSize: 0.8rem; }

.SocialMediaShareButton {
  display: inline-block;
  cursor: pointer; }
  .SocialMediaShareButton div {
    width: 35px !important;
    height: 35px !important; }
  .SocialMediaShareButton svg {
    width: 30px;
    height: 30px; }

.SocialMediaShareButton--grayScale svg {
  filter: grayscale(1); }

.Footer {
  padding: 2.5rem 0;
  color: var(--Footer-color);
  text-align: center;
  background-color: var(--Footer-backgroundColor); }
  .Footer .Footer__Copyright {
    font-size: var(--Footer__Copyright-fontSize); }
  .Footer .Footer__Sections__Header {
    font-size: var(--Footer__Sections__Header-fontSize); }
  .Footer a {
    color: var(--Footer-color);
    font-size: var(--Footer__Sections__Link-fontSize); }
  .Footer a:hover {
    color: var(--Footer__Sections__Link--hover-color); }
