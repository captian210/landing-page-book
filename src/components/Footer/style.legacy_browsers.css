:root {
  --Footer-color: #cccccc;
  --Footer-backgroundColor: #383d45;
  --Footer__Sections__Link--hover-color: #fed136;
  --Footer__Sections__Header-fontSize: 1.5rem;
  --Footer__Sections__Link-fontSize: 0.8rem;
  --Footer__Copyright-fontSize: 0.8rem; }

.SocialMediaShareButton {
  display: inline-block;
  cursor: pointer; }
  .SocialMediaShareButton div {
    width: 35px !important;
    height: 35px !important; }
  .SocialMediaShareButton svg {
    width: 30px;
    height: 30px; }

.SocialMediaShareButton--grayScale svg {
  filter: grayscale(1); }

.Footer {
  padding: 2.5rem 0;
  color: #cccccc;
  text-align: center;
  background-color: #383d45; }
  .Footer .Footer__Copyright {
    font-size: 0.8rem; }
  .Footer .Footer__Sections__Header {
    font-size: 1.5rem; }
  .Footer a {
    color: #cccccc;
    font-size: 0.8rem; }
  .Footer a:hover {
    color: #fed136; }
