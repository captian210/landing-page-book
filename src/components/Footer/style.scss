.Footer {
  padding: 2.5rem 0;
  color: $Footer-color;
  text-align: center;
  background-color: $Footer-backgroundColor;

  .Footer__Copyright{
    font-size: $Footer__Copyright-fontSize;
  }

  .Footer__Sections__Header{
    font-size: $Footer__Sections__Header-fontSize;
  }

  .Footer__Sections__Link{
    color: $Footer-color;
    font-size: $Footer__Sections__Link-fontSize;
  }
  .Footer__Sections__Link:hover{
    color: $Footer__Sections__Link--hover-color;
  }
}