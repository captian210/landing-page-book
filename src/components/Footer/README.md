<a style="float:right; margin-top: 30px;" target="_blank" href="https://github.com/front10/landing-page-book/edit/master/src/components/Footer/README.md"> <img width="15px;" src="https://assets-cdn.github.com/images/icons/emoji/unicode/270f.png"/> Edit on Github
</a>

# Footer

[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

## Footer

`@front10/landing-page-book/Footer` is used to create footers.

## How to use

1- Importing bootstrap style

```js
import 'bootstrap/dist/css/bootstrap.min.css';
```

2- Add following content to import the component:

```js
import Footer from '@front10/landing-page-book/dist/components/Footer';
import '@front10/landing-page-book/dist/components/Footer/style.css';
```

> Note: For including all components styles once you can use [a theme](https://github.com/front10/landing-page-book/wiki/Theming).

3- Put this code into jsx page:

```html
<Footer/>
```
<a target="_blank" href="https://codesandbox.io/s/x40y1npnw">
  <img alt="Edit Footer demo" src="https://codesandbox.io/static/img/play-codesandbox.svg">
</a>

You can use Footer component with children too:

```html
<Footer>
    <div className="row">
        <div className="col-sm-12 col-md">
            <Copyright text="Front10, Inc" showAllRightText={false}/>
        </div>
        <div className="col-sm-12 col-md mt-3 mt-md-0">
            <img src="https://front10.com/img/logos/logo-main.png" width={80}/>
        </div>
        <div className="col-sm-12 col-md mt-3 mt-md-0">
            <Social type="twitter" url="https://front10.com" gray/>
            <Social type="facebook" url="https://front10.com" gray/>
            <Social type="linkedin" url="https://front10.com" gray/>
        </div>
    </div>
</Footer>
```
<a target="_blank" href="https://codesandbox.io/s/14vnz2l3xq">
  <img alt="Edit Footer with childrens" src="https://codesandbox.io/static/img/play-codesandbox.svg">
</a>

## Example

```js
const sections = [
  {
    name: 'Features',
    sections: [
      {
        name: 'Cool stuff',
        url: 'https://front10.com'
      },
      {
        name: 'Random feature',
        url: 'https://front10.com'
      },
      {
        name: 'Team feature',
        url: 'https://front10.com'
      },
      {
        name: 'Stuff for developers',
        url: 'https://front10.com'
      },
      {
        name: 'Another one',
        url: 'https://front10.com'
      },
      {
        name: 'Last time',
        url: 'https://front10.com'
      }
    ]
  },
  {
    name: 'Resources',
    sections: [
      {
        name: 'Resource',
        url: 'https://front10.com'
      },
      {
        name: 'Resource name',
        url: 'https://front10.com'
      },
      {
        name: 'Another resource',
        url: 'https://front10.com'
      },
      {
        name: 'Final resource',
        url: 'https://front10.com'
      }
    ]
  },
  {
    name: 'About',
    sections: [
      {
        name: 'Team',
        url: 'https://front10.com'
      },
      {
        name: 'Locations',
        url: 'https://front10.com'
      },
      {
        name: 'Privacy',
        url: 'https://front10.com'
      },
      {
        name: 'Terms',
        url: 'https://front10.com'
      }
    ]
  }
];
```

## Properties:

| </br>Name     | </br>Type | </br>Summary                                                             |
| ------------- | --------- | ------------------------------------------------------------------------ |
| socialRounded | `Boolean` | Define if socials buttons are rounded. Default `true`                    |
| socialGray    | `Boolean` | Define if socials buttons are gray. Default `false`                      |
| socialUrl     | `String`  | Url to share in socials networks. Default `""`                           |
| copyright     | `String`  | Text of copyright. Default `""`                                          |
| socials       | `Array`   | Array of string with values of socials networks to display. Default `[]` |
| sections      | `Array`   | Sections to show . Default `[]`, see example section                     |

## Theming variables

| </br>Name                                 | </br>Summary                           | </br>Default |
| ----------------------------------------- | -------------------------------------- | ------------ |
| --Footer-color                            | Color of footer                        | `#cccccc`    |
| --Footer-backgroundColor                  | Background color of footer             | `#383d45`    |
| --Footer\_\_Sections\_\_Link--hover-color | Hover color of sections in footer      | `#fed136`    |
| --Footer\_\_Sections\_\_Header-fontSize   | Font size of sections header in footer | `1.5rem`     |
| --Footer\_\_Sections\_\_Link-fontSize     | Font size of sections links in footer  | `0.8rem`     |
| --Footer\_\_Copyright-fontSize            | Font size of copy right in footer      | `0.8rem`     |
| --Footer-backgroundImage                  | Footer background image                | `none`       |
