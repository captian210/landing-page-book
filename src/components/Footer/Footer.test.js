import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Footer from './Footer';

configure({ adapter: new Adapter() });

describe('Footer', () => {
  let props;
  let mounted;
  const footer = () => {
    if (!mounted) {
      mounted = mount(<Footer {...props} />);
    }
    return mounted;
  };
  describe('footer', () => {
    beforeEach(() => {
      props = {
        sections: [
          {
            name: 'Features',
            sections: [
              {
                name: 'Cool stuff',
                url: 'https://front10.com'
              },
              {
                name: 'Random feature',
                url: 'https://front10.com'
              },
              {
                name: 'Team feature',
                url: 'https://front10.com'
              },
              {
                name: 'Stuff for developers',
                url: 'https://front10.com'
              },
              {
                name: 'Another one',
                url: 'https://front10.com'
              },
              {
                name: 'Last time',
                url: 'https://front10.com'
              }
            ]
<<<<<<< HEAD
          },
          {
            name: 'Resources',
            sections: [
              {
                name: 'Resource',
                url: 'https://front10.com'
              },
              {
                name: 'Resource name',
                url: 'https://front10.com'
              },
              {
                name: 'Another resource',
                url: 'https://front10.com'
              },
              {
                name: 'Final resource',
                url: 'https://front10.com'
              }
            ]
          },
          {
            name: 'About',
            sections: [
              {
                name: 'Team',
                url: 'https://front10.com'
              },
              {
                name: 'Locations',
                url: 'https://front10.com'
              },
              {
                name: 'Privacy',
                url: 'https://front10.com'
              },
              {
                name: 'Terms',
                url: 'https://front10.com'
              }
            ]
          }
        ]
=======
          }]
>>>>>>> c8598f6ff33bbb153a69ae03fdc592f10e57fb05
      };
      mounted = undefined;
    });

    it('should render', () => {
      expect(mount(<footer {...props} />)).toMatchSnapshot();
    });

    it('always renders a div', () => {
      const divs = footer().find('footer');

      expect(divs.length).toBeGreaterThan(0);
    });

    it('should render sections', async () => {
      const ul = footer().find('ul');

      expect(ul.length).toBeGreaterThan(0);
    });
  });
});
