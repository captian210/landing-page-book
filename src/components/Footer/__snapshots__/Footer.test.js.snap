// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer footer should render 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": <footer />,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": Array [
      undefined,
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": <footer />,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
